pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -269,6 +278,133 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundHTTPRouteConfig(env *m
 	return out
 }
 
+//BuildVirtualHosts builds a list of virtual hosts for a specified routeConfig name.
```

Comment:
I assume this is used for the delta protocol ? Not sure I follow, my understanding was that Pilot will send the normal 
list of routes, and envoy will do an on-demand request for any new host.

Reply:
Yep, this is used for delta VHDS. In a delta RDS/VHDS environment, when the Envoy proxy sends the first RDS request, it should receive the routes with virtual hosts for every entry in the sidecar CRD. If/when the VHDS connection is opened, then Pilot should switch over to sending virtual host updates over VHDS while routes are sent over RDS. This function separates a method for VHDS to get the list of virtual hosts to return without all of the associated routes. This means that requests for additional routes can be made without including all of the virtual hosts that belong to it as well.